pkg/resources/oauth_integration.go

```diff
@@ -0,0 +1,281 @@
+package resources
+
+import (
+	"database/sql"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/pkg/errors"
+)
+
+var oauthIntegrationSchema = map[string]*schema.Schema{
+	"name": {
+		Type:        schema.TypeString,
+		Required:    true,
+		ForceNew:    true,
+		Description: "Specifies the name of the OAuth integration. This name follows the rules for Object Identifiers. The name should be unique among security integrations in your account.",
+	},
+	"oauth_client": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "Specifies the OAuth client type.",
+		ValidateFunc: validation.StringInSlice([]string{
+			"TABLEAU_DESKTOP", "TABLEAU_SERVER", "LOOKER",
+		}, false),
+	},
+	"oauth_issue_refresh_tokens": {
+		Type:        schema.TypeBool,
+		Optional:    true,
+		Description: "Specifies whether to allow the client to exchange a refresh token for an access token when the current access token has expired.",
+	},
+	"oauth_refresh_token_validity": {
+		Type:        schema.TypeInt,
+		Optional:    true,
+		Description: "Specifies how long refresh tokens should be valid (in seconds). OAUTH_ISSUE_REFRESH_TOKENS must be set to TRUE.",
```

Comment:
do we need `oauth_issue_refresh_tokens` or would the presence/absence of `oauth_refresh_token_validity` give us enough information?

Reply:
I see that keeping both would more closely match Snowflake's api, so we might want to keep as is

Reply:
Looks like `oauth_issue_refresh_tokens` is true by default, so you need it here if you want to set it to false. However, `oauth_refresh_token_validity` is optional too, it defaults to various values depending on the client (10 hours for Tableau Desktop, 90 days for Tableau Server, etc) - there will be cases where people omit both or specify only one.